ocaml/typing/typemod.ml

```diff
@@ -102,6 +102,7 @@ type error =
       old_arg_type : Global_module.Name.t option;
       old_source_file : Misc.filepath;
     }
+  | Duplicate_parameter_name of Global_module.Name.t
```

Comment:
This error is funny: it's only used by a `Misc.fatal_errorf`. This PR seems to assume that the build system won't generate wrong instances (as the duplication check is not complete anyway). Therefore, we don't need this error.

Reply:
That's only true because the code that calls it is a stub for unimplemented code. The next PR will have real code in `type_module_extension_aux` and it should produce type errors as usual. Not sure what you mean about the duplication check not being strict enough? I can add the check to `Global_module.create` if that would help.

Reply:
(decided to add the check to `Global_module.create`)

Reply:
Thank you for making the check complete.

I think if you rewrite `type_module_extension_aux` to:
```
      ignore (alias, sttn, env, smod);
     let modname = instance_name ~loc:smod.pmod_loc env glob in
      Misc.fatal_errorf "@[<hv>Unimplemented: instance identifier@ %a@]"
        Global_module.Name.print modname
```
it would be more clearer. That is, we want to separate "instance names are checked" and "the proper type checking hasn't been implemented". Previously I was confused because the two are together.